packages/core/src/tracing/cls.js

```diff
@@ -549,31 +552,50 @@ function skipExitTracing(options) {
       isActive: true,
       extendedResponse: false,
       skipParentSpanCheck: false,
-      skipIsTracing: false
+      skipIsTracing: false,
+      checkReducedSpan: false
     },
     options
   );
 
-  const parentSpan = getCurrentSpan();
+  let parentSpan = getCurrentSpan();
```

Comment:
```suggestion
  let parentSpan = getCurrentSpan();
  // If there is no active entry span, we fall back to the reduced span of the most recent entry span.
  // See comment in packages/core/src/tracing/clsHooked/unset.js#storeReducedSpan.
if (opts.checkReducedSpan && !parentSpan) {
  parentSpan = getReducedSpan();
}
```
The getCurrentSpan() is called twice in the condition. To avoid duplicate function calls, you could cache its result before checking for the reduced span: